### DISCUSSION TITLE
Creating enhanced opportunity for gns3 users view gns3 server logs in real-time

### DISCUSSION THREAD

**Discussion by josephmhiggins** [GNS3/gns3-server#2120] [General]
A discussion on handling the gns3 config files is over here: https://github.com/GNS3/gns3-server/discussions/2118

The last piece to solve is the gns3 logs. Gns3 version >= 3.0 is ignored for this discussion.
Given that gns3 users like free stuff, for GNS3 Windows users I am very optimistic that I can implement Solarwinds Event Log Forwarding and Solarwinds Kiwi SysLog server to allow gns3 users to view gns3 server logs in real-time.

I do not know how to do that yet. It is a high priority and I can probably figure out how to do it by end of 2022.

If gns3 users have a commercial grade syslog system, then they can use their own syslog server and Solarwinds Event Log Forwarding is still useful.

As far as creating a custom event log forwarder, I just read 2 days ago that you can use something in .NET to register a program as a "listener" to some kind of event, someway, somehow. In other words, I have no idea how I could register a powershell script to "listen" to the gns3 logs.

As far as the gns3 developers pushing log data out to some place easy to "listen" to:
1. that seems like a ton of work to me, and
2. even some expert engineers do not inspect syslog data to troubleshoot a problem; some expert engineers like to, for some reason, like to start from scratch when troubleshooting a problem. I am not getting involved with that type of discussion.

**Comment by eantowne** [comment]
On Linux, this would be a fairly easy solution. Windows is more problematic. I like the idea, I can see it being very useful for the small segment of the userbase that operates GNS3 at scale. As for enabling better troubleshooting on a single user (smallscale) environment, from my perspective grabbing the .log file is easier than pulling information out of Windows Event Viewer.

**Comment by josephmhiggins** [comment]
the gui for windows event viewer has not worked on my computer for the last 5 years.
I have spent between 8 - 12 hours to fix it through multiple restores of windows 10.
I have never fixed "Snap-in failed to load".
The event logs for windows are not easy to parse with powershell.
And, vmware only sends - i think - nat or dhcp information to windows event logs.
And, gns3 sends nothing to any event logs.
But....the gns3 developers should not send anything to windows event logs because it would be a major pain to parse it from a user standpoint so they should not developer any kind of framework that no gns3 user would try to access.

**Comment by josephmhiggins** [comment]
I will handle this on my own. I have this thread saved. This thread can be closed.
